swarm/src/lib.rs

```diff
@@ -349,11 +349,15 @@ where
 
 impl<TBehaviour> Unpin for Swarm<TBehaviour> where TBehaviour: NetworkBehaviour {}
 
+#[allow(deprecated)]
 impl<TBehaviour> Swarm<TBehaviour>
 where
     TBehaviour: NetworkBehaviour,
 {
     /// Builds a new `Swarm` with a provided executor.
+    #[deprecated(
+        note = "Use a builder chain `SwarmBuilder::with_executor(transport, behaviour, local_peer_id, executor).build()` instead."
```

Comment:
I think we can shorten this to:

```suggestion
        note = "Use `SwarmBuilder::with_executor` instead."
```

Reply:
I will replace it, however, the message I propose could be used as a direct replacement.

Reply:
> I will replace it, however, the message I propose could be used as a direct replacement.

That only works if users use these exact variable names. The only thing we can say for sure is the name of the function so that is what I'd like to point to.

The story would be different if Rust had support for codemods :)